crates/forge-runner/src/lib.rs

```diff
@@ -0,0 +1,491 @@
+use crate::fuzzer::RandomFuzzer;
+use crate::running::{run_fuzz_test, run_test};
+use crate::test_case_summary::TestCaseSummary;
+use crate::test_crate_summary::TestCrateSummary;
+use anyhow::{anyhow, Context, Result};
+use cairo_lang_runner::SierraCasmRunner;
+use cairo_lang_sierra::ids::ConcreteTypeId;
+use cairo_lang_sierra::program::{Function, Program};
+use cairo_lang_sierra_to_casm::metadata::MetadataComputationConfig;
+use cairo_lang_utils::ordered_hash_map::OrderedHashMap;
+use camino::Utf8PathBuf;
+use console::style;
+use futures::stream::FuturesUnordered;
+use futures::StreamExt;
+use once_cell::sync::Lazy;
+use scarb_artifacts::StarknetContractArtifacts;
+use smol_str::SmolStr;
+use starknet::core::types::BlockId;
+use std::collections::HashMap;
+use std::sync::Arc;
+use test_collector::FuzzerConfig;
+use test_collector::TestCase;
+use test_collector::{ForkConfig, LinkedLibrary, RawForkParams};
+use tokio::sync::mpsc::{channel, Sender};
+use tokio::task::JoinHandle;
+use url::Url;
+
+mod fuzzer;
+mod running;
+pub mod test_case_summary;
+pub mod test_crate_summary;
+mod test_execution_syscall_handler;
+
+pub const CACHE_DIR: &str = ".snfoundry_cache";
+
+pub static BUILTINS: Lazy<Vec<&str>> = Lazy::new(|| {
+    vec![
+        "Pedersen",
+        "RangeCheck",
+        "Bitwise",
+        "EcOp",
+        "Poseidon",
+        "SegmentArena",
+        "GasBuiltin",
+        "System",
+    ]
+});
+
+#[derive(Debug, PartialEq, Clone)]
+pub struct ForkTarget {
+    name: String,
+    params: RawForkParams,
+}
+
+impl ForkTarget {
+    #[must_use]
+    pub fn new(name: String, params: RawForkParams) -> Self {
+        Self { name, params }
+    }
+
+    #[must_use]
+    pub fn name(&self) -> &str {
+        &self.name
+    }
+
+    #[must_use]
+    pub fn params(&self) -> &RawForkParams {
+        &self.params
+    }
+}
```

Comment:
This impl is not really useful, can't we just mark the fields `pub` and avoid adding `new` method if it doesn't wrap any logic/add any context?

Reply:
Let's discuss offline

Reply:
#1123